src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AzureAttestationBasedEnclaveProvider.cs

```diff
@@ -332,7 +332,7 @@ private static string GetInnerMostExceptionMessage(Exception exception)
         // It also caches that information for 1 day to avoid DDOS attacks.
         private OpenIdConnectConfiguration GetOpenIdConfigForSigningKeys(string url, bool forceUpdate)
         {
-            OpenIdConnectConfiguration openIdConnectConfig = OpenIdConnectConfigurationCache[url] as OpenIdConnectConfiguration;
+            OpenIdConnectConfiguration openIdConnectConfig = OpenIdConnectConfigurationCache.Get<OpenIdConnectConfiguration>(url);
```

Comment:
Here, can you test if the code can be change to below:
```C#
ConfigurationManager<OpenIdConnectConfiguration> configManager = new(url, new OpenIdConnectConfigurationRetriever());
OpenIdConnectConfiguration openIdConnectConfig = configManager.GetConfigurationAsync().Result;
```

Reply:
I do not understand what to test since the MemoryCache does not have GetConfigurationAsync function.

Reply:
This seems like beyond the scope of this change.

Reply:
The `Get` will add `Microsoft.Extensions.Caching.Abstractions` dependency, I am trying to reduce extensions usage. I think What we need to know is if there is a workaround to avoid using extensions.